tagstudio/src/qt/widgets/thumb_renderer.py

```diff
@@ -746,6 +757,46 @@ def _video_thumb(self, filepath: Path) -> Image.Image:
             )
         return im
 
+    def _pdf_thumb(self, filepath: Path, size: int) -> Image.Image:
+        file: QFile = QFile(filepath)
+        was_open_success: bool = file.open(
+            QIODeviceBase.OpenModeFlag.ReadOnly, QFileDevice.Permission.ReadUser
+        )
+        if not was_open_success:
+            logging.error(
+                f"[ThumbRenderer][PDF][ERROR]: Couldn't open pdf-file {filepath.name}"
+            )
+            return None
+        document: QPdfDocument = QPdfDocument()
+        document.load(file)
+        # Transform page_size in points to pixels with proper aspect ratio
+        page_size: QSizeF = document.pagePointSize(0)
+        ratio_hw: float = page_size.height() / page_size.width()
+        if ratio_hw >= 1:
+            page_size *= size / page_size.height()
+        else:
+            page_size *= size / page_size.width()
+        # Enlarge image to improve image downscaling (kind of arbitrary)
+        page_size *= 2.5
+        # Render image with no anti-aliasing for speed
+        render_options: QPdfDocumentRenderOptions = QPdfDocumentRenderOptions()
+        render_options.setRenderFlags(
+            QPdfDocumentRenderOptions.RenderFlag.TextAliased
+            | QPdfDocumentRenderOptions.RenderFlag.ImageAliased
+            | QPdfDocumentRenderOptions.RenderFlag.PathAliased
+        )
+        # Convert QImage to PIL Image
+        rendered_qimage: QImage = document.render(0, page_size.toSize(), render_options)
+        buffer: QBuffer = QBuffer()
+        buffer.open(QBuffer.OpenModeFlag.ReadWrite)
+        rendered_qimage.save(buffer, "PNG")
+        pil_image: Image.Image = Image.open(BytesIO(buffer.buffer().data()))
+        buffer.close()
```

Comment:
can you use a context manager instead? 
I don't know `QBuffer` so I cant make an inline suggestion, I'm open to research this and help you out.

Reply:
AFAIK Qt for Python is directly generate from the C++ code without adding python-like features. I could add a context manager for the buffer but I do not think that is a good idea and would generate too much unnecessary code.
However, I will definitely put it in a "fake context manager" with try-except to close the buffer even with errors occuring.